### PR TITLE
add symmetric crypting to GPG and Password Dialog

### PR DESCRIPTION
**Pull request by jreidinger**
trello: https://trello.com/c/TQPehs2U/2015-8-refactor-the-autoyast-profile-loading-process

goal: adding symetric gpg to gpg module, so it can be used as backend when autoyast profile is encrypted. Also add shared PasswordDialog.

### CODE REVIEW

**Review by imobachgs — APPROVED** [review]
LGTM

File: library/gpg/test/gpg_test.rb [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+# find current contact information at www.suse.com.
+
+require_relative "test_helper"
+require "tempfile"
+
+Yast.import "GPG"
+
+describe Yast::GPG do
+  let(:file_content) do
+    "-----BEGIN PGP MESSAGE-----
```
Comment (lslezak) [inline library/gpg/test/gpg_test.rb:28]: Um, why not having it as a plain data file? That would avoid that save/remove `around` block...

Like we have https://github.com/yast/yast-yast2/tree/master/library/packages/test/y2packager/zypp/test_repo/repodata or https://github.com/yast/yast-yast2/tree/master/library/packages/test/data/rpm

**Review by lslezak — COMMENTED** [review]
Data file in a test?

File: library/gpg/test/gpg_test.rb (hunk 1 above)
Comment (jreidinger) [reply library/gpg/test/gpg_test.rb:28]: well, I consider it small enough, but moving is ok for me

File: library/general/src/lib/ui/password_dialog.rb [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+      @label = label
+
+      super()
+    end
+
+    def dialog_content
+      res = VBox(
+        Left(Heading(@label)),
+        HBox(
+          MinWidth(18, Label(_("Password:"))),
```
Comment (lslezak) [inline library/general/src/lib/ui/password_dialog.rb:51]: We already have [Label.Password](https://github.com/yast/yast-yast2/blob/master/library/general/src/modules/Label.rb#L321) and [Label.ConfirmPassword](https://github.com/yast/yast-yast2/blob/master/library/general/src/modules/Label.rb#L327), used like [here](https://github.com/yast/yast-users/blob/master/src/include/users/dialogs.rb#L517-L518).

**Review by lslezak — COMMENTED** [review]
Label.Password and Label.ConfirmPassword

File: library/gpg/test/gpg_test.rb [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+  describe "#encrypted_symmetric" do
+    let(:file_content) { "test\n" }
+
+    it "will create output file that is encrypted" do
+      Tempfile.open do |f|
+        path = f.path
+        f.close
+        f.unlink
+        described_class.encrypt_symmetric(@path, path, "test")
+        expect(described_class.encrypted_symmetric?(path)).to eq true
```
Comment (lslezak) [inline library/gpg/test/gpg_test.rb:90]: The created temporary file is kept on the disk, it should be removed at the end.

File: library/gpg/test/gpg_test.rb [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+  end
+
+  describe "#encrypted_symmetric" do
+    let(:file_content) { "test\n" }
+
+    it "will create output file that is encrypted" do
+      Tempfile.open do |f|
+        path = f.path
+        f.close
+        f.unlink
```
Comment (lslezak) [inline library/gpg/test/gpg_test.rb:88]: There is a small race condition between unlinking the temporary file and writing it. In theory some other process could create a file with the same name in the meantime...

Does GPG has force option to overwrite an existing file? Or output the encrypted file on stdout and redirect it with `>` to the target file...

**Review by lslezak — COMMENTED** [review]
Small test issues...

**Review by lslezak — APPROVED** [review]

**Pull request merged**